### PR TITLE
add formField.checkBox.pad as an option to be handeled in formField

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
adds `formField.checkBox.pad` this will add `pad` to checkbox from theme.
#### Where should the reviewer start?
FormField
#### What testing has been done on this PR?
set pad value to formField.checkBox.pad
#### How should this be manually tested?
set pad value to formField.checkBox.pad
Custom story
#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
needed for theme
https://github.com/grommet/grommet-theme-hpe/pull/269
#### Do the grommet docs need to be updated?
yes I will add a PR if this is the direction we go
#### Should this PR be mentioned in the release notes?
sure
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/FormField/FormField.js [hunk 1]
```diff
@@ -229,6 +230,19 @@ const FormField = forwardRef(
               focusIndicator: false,
```
Comment (taysea) [inline src/js/components/FormField/FormField.js:230]:
```suggestion
              focusIndicator: false,
              pad: 'CheckBox'.indexOf(child.type.displayName) !== -1 ? formFieldTheme?.checkBox?.pad : undefined,
```

Does doing this work? And then removing the new 'if' statement below? Just trying to minimize repeated logic if possible

File: src/js/components/FormField/FormField.js [hunk 2]
```diff
@@ -220,6 +220,7 @@ const FormField = forwardRef(
           if (
             child &&
             child.type &&
+            'CheckBox'.indexOf(child.type.displayName) === -1 &&
```
Comment (taysea) [inline src/js/components/FormField/FormField.js:223]: When testing my other comment, we would need to remove this line.

File: src/js/components/FormField/FormField.js (hunk 1 above)
Comment (britt6612) [reply src/js/components/FormField/FormField.js:230]: let me test on my side!

Comment (britt6612) [reply src/js/components/FormField/FormField.js:229]: worked good!

**Comment by taysea** [comment]
Can we add a unit test for this?

**Review by taysea — APPROVED** [review]
LGTM, thanks for adding the test. I tried the behavior locally and looks good.

**Comment by britt6612** [comment]
> LGTM, thanks for adding the test. I tried the behavior locally and looks good.

thank you for the feedback 😄

**Review by ericsoderberghp — APPROVED** [review]

File: src/js/themes/base.d.ts [hunk 3]
```diff
@@ -782,6 +782,9 @@ export interface ThemeType {
       margin?: MarginType;
       pad?: PadType;
     };
+    checkBox: {
+      pad: PadType;
```
Comment (lunika) [inline src/js/themes/base.d.ts:786]: Should not be optional like other keys ?

Comment (lunika) [reply src/js/themes/base.d.ts:786]: Same for `CheckBox` key.

Comment (britt6612) [reply src/js/themes/base.d.ts:786]: yes you are correct pushing up a fix now sorry for any inconvenience. Thanks for the catch!

**Pull request merged**